underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java

```diff
@@ -199,32 +206,69 @@ public static S3AUnderFileSystem createInstance(AlluxioURI uri,
       clientConf.setSignerOverride(conf.get(PropertyKey.UNDERFS_S3_SIGNER_ALGORITHM));
     }
 
-    AmazonS3Client amazonS3Client = new AmazonS3Client(credentials, clientConf);
-
-    // Set a custom endpoint.
-    if (conf.isSet(PropertyKey.UNDERFS_S3_ENDPOINT)) {
-      amazonS3Client.setEndpoint(conf.get(PropertyKey.UNDERFS_S3_ENDPOINT));
+    AmazonS3ClientBuilder clientBuilder = AmazonS3ClientBuilder
+        .standard()
+        .withCredentials(credentials)
+        .withClientConfiguration(clientConf)
+        .withPathStyleAccessEnabled(Boolean.parseBoolean(conf
```

Comment:
I think we don't want to override whatever is the default if `withPathStyleAccessEnabled` is not called.
So ideally, not even call `withPathStyleAccessEnabled` when `Boolean.parseBoolean(conf.get(PropertyKey.UNDERFS_S3_DISABLE_DNS_BUCKETS)` is false.

Reply:
Sounds good! Only set when alluxio conf is set that property, PTAL, thanks